### DISCUSSION TITLE
read  a tree sequence into msprime ?

### DISCUSSION THREAD

**Discussion by eldonb** [tskit-dev/msprime#2274] [Q&A] [answered]
hi all, 
was wondering if it is possible to  read   tree  sequence data  into msprime  ?

I am  working on individual-based simulations that generate ancestral trees in two ways,  and would  like to 
plot some example  trees  using the tskit  draw_svg  facility, and possibly for comparison with  output from msprime; 
so I have my own code for generating trees  and once converted  to tree sequence format  read them into  msprime ?

a simple  example -  for a single non-recombining locus I have an output (in compact form) 
0 1 2
i   i  j
k k 
 
meaning that  I label the sampled leaves as 0,1,2;  after (say) r generations the ancestors of leaves  0 and 1 merge, and after further (say) s 
generations  the remaining two ancestors merge; the tree sequence  would then be (I reckon)   <3,3, 4, 4, 0>  
with node times (0,0,0, r, r + s) 

so was wondering if it is possible to read tree sequences generated outside msprime into msprime ?

best
bjarki

**Comment by eldonb** [comment]
update:  was  viewing tskit and msprime  as one and the same thing - 
maybe a more accurate question is : 
are there examples  showing how to generate  tree sequence data from scratch  (even a made-up example, 
not an output from any  simulation software, and just in the simplest case of a single non-recombining locus, and only the bare-bones ancestral relations)  that the tskit library can  process,  in particular plot trees ?

**Comment by GertjanBisschop** [comment, marked as answer]
Hi @eldonb,

Good starting points might be [editing tables](https://tskit.dev/tutorials/tables_and_editing.html) and [building simulator with tskit](https://tskit.dev/tutorials/forward_sims.html). Especially the first link will highlight how you can modify/build the tables that underlie any tree sequence yourself. Once you get to grips with the fundamental building blocks it is relatively easy to code up your own example tree sequences and plot them.
